tests/e2e/test_queries.py

```
@@ -0,0 +1,48 @@
+    query = query_file.read_text()
+    server.add_query(query)
+
+    csv_path = dataset["base"] / dataset["csv"]
+    events, delays_ns = parse_csv(csv_path, stream_info)
+
+    has_timing = any(d > 0 for d in delays_ns)
+    if has_timing:
+        # Replay events with inter-event timing delays to match offline
+        # binary behavior (required for quarantine policies like
+        # BOUNDED_TIME that depend on wall-clock event arrival spacing).
+        for event, delay_ns in zip(events, delays_ns):
```

Ruff (0.15.2)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)